src/utils/permissions.ts

```diff
@@ -191,8 +112,134 @@ export async function hasChannelPermissions(
   if (permissions.every((perm) => allowedPermissions.has(perm))) return true;
 
   // Some permission was not explicitly allowed so we default to checking role perms directly
-  const hasPerms = await botHasPermission(guild.id, permissions);
-  return hasPerms;
+  return await calculateServerPermissions(memberID, guild.id, permissions);
```

Comment:
Don't do return await
```suggestion
  return calculateServerPermissions(memberID, guild.id, permissions);
```